src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Completion/CompletionHandler.cs

```diff
@@ -0,0 +1,217 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Options;
+using System.Net.WebSockets;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig.Features
+{
+    [ExportStatelessLspService(typeof(CompletionHandler), ProtocolConstants.EditorConfigLanguageContract), Shared]
+    [Method(Methods.TextDocumentCompletionName)]
+    internal class CompletionHandler : IRequestHandler<CompletionParams, CompletionList?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CompletionHandler()
+        {
+        }
+
+        public bool MutatesSolutionState => false;
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(CompletionParams request) => request.TextDocument;
+
+        public async Task<CompletionList?> HandleRequestAsync(CompletionParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.AdditionalDocument;
+            Contract.ThrowIfNull(document);
+
+            var workspace = context.Solution?.Workspace;
+            Contract.ThrowIfNull(workspace);
+
+            Contract.ThrowIfNull(request.Context);
+
+            var filePath = document.FilePath;
+            Contract.ThrowIfNull(filePath);
+
+            var optionSet = workspace.Options;
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var offset = text.Lines.GetPosition(ProtocolConversions.PositionToLinePosition(request.Position));
+            var textInLine = text.Lines.GetLineFromPosition(offset).ToString();
+
+            var settingsAggregator = workspace.Services.GetRequiredService<ISettingsAggregator>();
+
+            var codeStyleProvider = settingsAggregator.GetSettingsProvider<CodeStyleSetting>(filePath);
+            var whitespaceProvider = settingsAggregator.GetSettingsProvider<WhitespaceSetting>(filePath);
+            var analyzerProvider = settingsAggregator.GetSettingsProvider<AnalyzerSetting>(filePath);
+
+            var codeStyleSnapshot = codeStyleProvider?.GetCurrentDataSnapshot();
+            var whitespaceSnapshot = whitespaceProvider?.GetCurrentDataSnapshot();
+            var analyzerSnapshot = analyzerProvider?.GetCurrentDataSnapshot();
+
+            // Check if the user has written a name and not a value
+            if (textInLine.Contains(' ') && !textInLine.Contains('='))
+            {
+                return null;
+            }
+
+            // Check if we need to show the name of the setting or the values
+            if (textInLine.Contains('='))
+            {
+                if (textInLine.Count(c => c == '=') > 1 || textInLine.Count(c => c == ':') > 1)
+                {
+                    return null;
+                }
+
+                var settingName = textInLine.Split('=').FirstOrDefault().Trim();
```

Comment:
no need for FirstOrDefault.  You already ensured that there was a `=` in the line.  so split must give back at least 2 items.